### PR TITLE
Second pass on migration guide

### PR DESCRIPTION
**Pull request by StephenBarlow**

### CODE REVIEW

File: docs/source/migration.mdx [hunk 1]
```diff
@@ -4,31 +4,30 @@ title: Migrating to Apollo Server 3
...
 
-**Apollo Server 3 is generally available.** The focus of this major-version release is to provide a lighter, nimbler core library as a foundation for future features and improved extensibility.
+Apollo Server 3 is generally available. The focus of this major-version release is to provide a lighter, nimbler core library as a foundation for future features and improved extensibility.
 
-**Many Apollo Server 2 users don't need to make any code changes to upgrade to Apollo Server 3**, especially if you use the "batteries-included" `apollo-server` library (as opposed to a [middleware-specific library](./integrations/middleware/)).
+**Many Apollo Server 2 users don't need to make any code changes to upgrade to Apollo Server 3!** This is especially likely if you use the "batteries-included" `apollo-server` library (as opposed to a [middleware-specific library](./integrations/middleware/)).
 
-This document explains which features _do_ require code changes and how to make them.
+This document explains which features _do_ require code changes and how to make them. If you encounter issues while migrating, please [let us know in the forums](https://community.apollographql.com/).
```
Comment (glasser) [inline docs/source/migration.mdx:11]: I think filing an issue would be fine for now. I'll be better at observing and responding on GitHub.

File: docs/source/migration.mdx [hunk 2]
```diff
@@ -162,17 +161,17 @@ import typeDefs from "./typeDefs";
...
+To use `graphql-upload` with Apollo Server 3, see the [documentation on enabling file uploads in Apollo Server](../data/file-uploads/). Note that if your project currently uses uploads with the "batteries-included" `apollo-server` package, you must first [swap to `apollo-server-express`](./integrations/middleware/#swapping-out-apollo-server).
 
 ### GraphQL Playground
 
-By default, Apollo Server 2 serves the (now-[retired](https://github.com/graphql/graphql-playground/issues/1143)) [GraphQL Playground](https://github.com/graphql/graphql-playground) IDE from its base URL (unless it's running in production). You can override this default behavior with the `playground` option (for example, `playground:true` enables GraphQL Playground even in production).
+By default, Apollo Server 2 serves the (now-[retired](https://github.com/graphql/graphql-playground/issues/1143)) [GraphQL Playground](https://github.com/graphql/graphql-playground) IDE from its base URL (except in production).
```
Comment (glasser) [inline docs/source/migration.mdx:172]: I think it's helpful to be explicit that `playground` used to be a constructor option and that it is removed.

I think it's helpful to be explicit that the old behavior was "playground in non-prod, nothing in prod" whereas the new behavior is "landing page always; landing page links to Sandbox in non-prod". I don't think your text maintains that detail.

File: docs/source/migration.mdx [hunk 3]
```diff
@@ -442,11 +441,15 @@ See [GraphQL Playground](#graphql-playground).
...
 
 ### Exports from `graphql-tools`
 
-Apollo Server 2 exported every symbol exported by [`graphql-tools`](https://www.graphql-tools.com/) v4. If you're importing any of the following symbols from an Apollo Server package, you should instead run `npm install graphql-tools@4.x` and import the symbol from `graphql-tools` instead. Alternatively, read the [GraphQL Tools docs](https://www.graphql-tools.com/docs/introduction) and find out which `@graphql-tools/subpackage` the symbol is exported from in more modern versions of GraphQL Tools.
+Apollo Server 2 exports every symbol exported by [`graphql-tools`](https://www.graphql-tools.com/) v4. If you're importing any of the following symbols from an Apollo Server package, you should instead run `npm install graphql-tools@4.x` and import the symbol from `graphql-tools`.
+
+> Alternatively, read the [GraphQL Tools docs](https://www.graphql-tools.com/docs/introduction) and find out which `@graphql-tools/subpackage` the symbol is exported from in more modern versions of GraphQL Tools.
+
+<ExpansionPanel title="Click to view symbols">
```
Comment (glasser) [inline docs/source/migration.mdx:452]: While the lack of an expansion panel does make the document longer, it also means you can immediately search for the name of the symbol involved in the error you're seeing upon upgrade and get a result. With the expansion panel you search for the symbol and get no results. I think it's better without. Maybe you have some split-the-difference way out of this though?

File: docs/source/migration.mdx [hunk 4]
```diff
@@ -620,36 +626,61 @@ class MyDataSource extends RESTDataSource {
...
+new ApolloError(message, code, {extensions: {key: 'value'}})
+```
+
+In Apollo Server 3, only the first line above is supported. If you try to use the second line, the constructor throws an error. Before upgrading, replace any code using the second form with the first form.
+
+#### Reading error extensions
+
+In Apollo Server 2, an error extension `foo` is present in two locations on an `ApolloError` object `error`: `error.foo` _and_ `error.extensions.foo`.
+
+In Apollo Server 3, it only shows up in one location: on `error.extensions.foo`. See [the PR](https://github.com/apollographql/apollo-server/pull/5294) for more details.
```
Comment (glasser) [inline docs/source/migration.mdx:681]: You removed the part about where the extensions show up in serialized JSON responses, which was weirdly different from where they show up on the JS object (this odd `extensions.exception` thing).  (And most people who care about this stuff are concerned about the JSON response, ie what they see in their client)

**Review by glasser — CHANGES_REQUESTED** [review]

File: docs/source/migration.mdx (hunk 3 above)
Comment (StephenBarlow) [reply docs/source/migration.mdx:452]: Don't think I do but I respect the use case. I'll remove the expander

File: docs/source/migration.mdx (hunk 4 above)
Comment (StephenBarlow) [reply docs/source/migration.mdx:681]: Ah yep missed the distinctions between the two, was clearly reading too fast

**Review by glasser — APPROVED** [review]

**Pull request merged**